src/structures.jl

```diff
@@ -18,19 +18,27 @@ mutable struct PrimalDualMap{T}
     primal_con_dual_con::Dict{CI, CI}
     primal_con_constants::Dict{CI, Vector{T}}
 
+    primal_parameter::Dict{VI, VI} # in the future we could have Parameters, as in ParameterJuMP
+    primal_var_dual_quad_slack::Dict{VI, VI}
+
     function PrimalDualMap{T}() where T
         return new(Dict{VI, CI}(),
                    Dict{CI, Vector{VI}}(),
                    Dict{CI, CI}(),
-                   Dict{CI, Vector{T}}())
+                   Dict{CI, Vector{T}}(),
+                   Dict{VI, VI}(),
+                   Dict{VI, VI}(),
+                   )
     end
 end
 
 function is_empty(primal_dual_map::PrimalDualMap{T}) where T
     if isempty(primal_dual_map.primal_var_dual_con) &&
```

Comment:
you could return the result of the condition directly
`return isempty().. && isempty(..) &&`